src/document/doc-printer.js

```diff
@@ -6,11 +6,16 @@ const getStringWidth = require("../utils/get-string-width.js");
 const { fill, cursor, indent } = require("./doc-builders.js");
 const { isConcat, getDocParts } = require("./doc-utils.js");
 
-/** @type {Record<symbol, typeof MODE_BREAK | typeof MODE_FLAT>} */
+/** @typedef {typeof MODE_BREAK | typeof MODE_FLAT} Mode */
+/** @typedef {{ ind: any, doc: any, mode: Mode }} Command */
+
+/** @type {Record<symbol, Mode>} */
 let groupModeMap;
 
-const MODE_BREAK = 1;
-const MODE_FLAT = 2;
+// prettier-ignore
+const MODE_BREAK = /** @type {const} */ (1);
+// prettier-ignore
+const MODE_FLAT = /** @type {const} */ (2);
```

Comment:
Without these `@type {const}` casts, variables like this
```js
const next = { ind, mode: MODE_FLAT, doc: doc.contents };`
```
are inferred as incompatible with the `Command` type.

`// prettier-ignore` is needed because Meriyah doesn't support parenthesized expressions.